### PR TITLE
Stats

### PR DESCRIPTION
**Pull request by mitya52**
* stats code moved here
* sqlalchemy as DB backend
* docker compose file to run refact and postgres images

### CODE REVIEW

File: self_hosting_machinery/dashboard_service/tests/insert_fake_tele_data.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+    {"language": "Dart", "tags": ["website", "plugins"], "extension": ".dart"},
+    {"language": "Scala", "tags": ["enterprise"], "extension": ".scala"},
+    {"language": "Perl", "tags": ["self-host"], "extension": ".pl"},
+    {"language": "Haskell", "tags": ["self-host"], "extension": ".hs"},
+    {"language": "Lua", "tags": ["self-host"], "extension": ".lua"}
+]
+
+
+@dataclass
+class FakeUser:
```
Comment (reymondzzzz) [inline self_hosting_machinery/dashboard_service/tests/insert_fake_tele_data.py:44]: MockUser. It's mock testing, not fake testing

File: self_hosting_machinery/dashboard_service/utils.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    extra: Dict
+
+
+class NoDataInDatabase(Exception):
+    pass
+
+
+def retrieve_all_data_tables(stats_service: StatisticsService) -> StatsDataTables:
+    extra = {}
+    network_df = pd.DataFrame(stats_service.network_select_all())
```
Comment (reymondzzzz) [inline self_hosting_machinery/dashboard_service/utils.py:25]: We don't need all. We need for period, for last year from now

File: self_hosting_machinery/webgui/selfhost_database.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+
+class RefactDatabase:
+
+    def __init__(self):
+        # NOTE: this is a hack to wait for a db to be ready
+        self._session: Optional[Session] = None
+        while True:
+            try:
+                with DisableLogger():
```
Comment (reymondzzzz) [inline self_hosting_machinery/webgui/selfhost_database.py:125]: What the different between Cassandra and PG here? Cassandra has models too, and there is while True loop for connection too? I don't understand... In cloud version we will need cassandra's keyspaces for teams. Does PG have that?

File: docker-compose.yml [hunk 4]
```diff
@@ -1,15 +1,34 @@
 version: "3.9"
+
 services:
-  refact_self_hosted:
+  refact:
     image: smallcloud/refact_self_hosting:latest
     volumes:
       - perm_storage:/perm_storage
-  nginx:
```
Comment (reymondzzzz) [inline docker-compose.yml:7]: why did u delete nginx, but didn't delete nginx.conf?

**Review by reymondzzzz — CHANGES_REQUESTED** [review]

File: self_hosting_machinery/webgui/selfhost_database.py (hunk 3 above)
Comment (mitya52) [reply self_hosting_machinery/webgui/selfhost_database.py:145]: keyspaces are designed for another purposes
using keyspaces for "team dbs" is a huge design mistake imo
anyway I went back to using cassandra as db

File: self_hosting_machinery/dashboard_service/utils.py (hunk 2 above)
Comment (mitya52) [reply self_hosting_machinery/dashboard_service/utils.py:25]: @valaises is the author of this code, he can fix this

**Comment by valaises** [comment]
@mitya52 
webgui does not start without a database. Is it expected?

**Review by valaises — APPROVED** [review]
seems to be working. However, it won't start with no database

**Comment by valaises** [comment]
@mitya52 issue with CAPS when sending stats:
```
/Users/kot/.cache/refact/telemetry/compressed/20231214-133628-net.json
to url
/stats/telemetry-basic
2023-12-14T12:36:28.514579Z ERROR refact_lsp::telemetry::basic_transmit:74: telemetry send failed: telemetry send failed: builder error: relative URL without a base
```
please, take a look

**Review by reymondzzzz — APPROVED** [review]

**Comment by mitya52** [comment]
cassandra instead of sqlalchemy

**Pull request merged**